app/controllers/katello/api/v2/environments_controller.rb

```diff
@@ -93,11 +94,17 @@ def create
     param :organization_id, :number, :desc => N_("name of the organization")
     param :new_name, String, :desc => N_("new name to be given to the environment")
     param :description, String, :desc => N_("description of the environment")
+    param :registry_name_pattern, String, :desc => N_("pattern for container image names")
     def update
-      fail HttpErrors::BadRequest, _("Can't update the '%s' environment") % "Library" if @environment.library?
       update_params = environment_params
+      fail HttpErrors::BadRequest, _("Can't update the '%s' environment") % "Library" if @environment.library? && update_params.empty?
       update_params[:name] = params[:environment][:new_name] if params[:environment][:new_name]
       @environment.update_attributes!(update_params)
+      if update_params[:registry_name_pattern]
+        #???? should task be returned? what would be returned when async_task not called?
+        #???? optional param to make this a sync_task?
```

Comment:
I think i would return the task here.  Technically the env is updated, its just further work that can be tracked

Reply:
@jlsherrill Should I change this to call update_attributes! in the plan() of the task so that there is always a task to return? (Is plan() the right place to do that?